### PR TITLE
Bug 844717 - Add Thai input method for Gaia. Picked up previous patch an...

### PR DESCRIPTION
**Pull request by gpgreen**
Squashed commits

### CODE REVIEW

**Comment by try-server-hook** [comment]
[gpgreen Greg Green (gpgreen)](https://github.com/gpgreen) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=639fb250eee7)

File: apps/keyboard/js/layouts/th.js [hunk 1]
```diff
@@ -0,0 +1,134 @@
+Keyboards.th = {
+  label: 'Thai',
+  shortLabel: 'Th',
+  menuLabel: 'ไทย', /*ไทย*/
+  secondLayout: true,
+  basicLayoutKey: 'กขค', /*กขค*/
+  imEngine: 'latin',
+  types: ['text', 'url', 'email'],
+  width: 11,
+  keys: [
```
Comment (RudyLu) [inline apps/keyboard/js/layouts/th.js]: we already change the format of the layout definition to allow it contain several pages, see the following file as an example.
https://github.com/mozilla-b2g/gaia/blob/master/apps/keyboard/js/layouts/gl.js#L8

File: apps/keyboard/js/layouts/th.js [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+    'อ': 'ฮ',
+    'ิ': 'ฺ',
+    'ื': '์',
+    'ท': '?',
+    'ม': 'ฒ',
+    'ใ': 'ฬ',
+    'ฝ': 'ฦ',
+    'ผ': '('
+  },
+  alternateLayout: {
```
Comment (RudyLu) [inline apps/keyboard/js/layouts/th.js]: This is not correct, see the above comment for how to define "pages".

File: apps/keyboard/js/layouts/th.js [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+          { value: '!' }, { value: '\"' }, { value: "'" }, { value: ':' },
+          { value: ';' }, { value: '/' }, { value: '?' },
+          { value: '⌫', ratio: 1.5, keyCode: KeyEvent.DOM_VK_BACK_SPACE }
+        ], [
+          { value: '&nbsp', ratio: 8, keyCode: KeyboardEvent.DOM_VK_SPACE },
+          { value: '↵', ratio: 2, keyCode: KeyEvent.DOM_VK_RETURN }
+        ]
+      ]
+  },
+  symbolLayout: {
```
Comment (RudyLu) [inline apps/keyboard/js/layouts/th.js]: same here, this is not correct, see the above comment for how to define "pages".

File: apps/keyboard/js/layouts/th.js [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+  menuLabel: 'ไทย', /*ไทย*/
+  secondLayout: true,
+  basicLayoutKey: 'กขค', /*กขค*/
+  imEngine: 'latin',
+  types: ['text', 'url', 'email'],
+  width: 11,
+  keys: [
+    [
+      { value: 'ๆ' }, { value: 'ภ' }, { value: 'ถ' }, { value: 'ุ' },
+      { value: 'ู' } , { value: 'ึ' }, { value: 'ค' } , { value: 'ต' },
```
Comment (RudyLu) [inline apps/keyboard/js/layouts/th.js]: extra space before `,`, please remove them.

File: apps/keyboard/js/layouts/th.js [hunk 5]
```diff
@@ -0,0 +1,134 @@
...
+      { value: 'จ' }, { value: 'ข' }, { value: 'ช' }
+    ], [
+      { value: 'ไ' }, { value: 'ำ' }, { value: 'พ' }, { value: 'ะ' },
+      { value: 'ั' } , { value: 'ี' }, { value: 'ร' }, { value: 'น' },
```
Comment (RudyLu) [inline apps/keyboard/js/layouts/th.js]: extra space to be removed.

File: apps/keyboard/js/layouts/th.js [hunk 6]
```diff
@@ -0,0 +1,134 @@
...
+      { value: 'ย' }, { value: 'บ' }, { value: 'ล' }
+    ], [
+      { value: 'ฟ' }, { value: 'ห' }, { value: 'ก' }, { value: 'ด' },
+      { value: 'เ' } , { value: '้' }, { value: '่' }, { value: 'า' },
```
Comment (RudyLu) [inline apps/keyboard/js/layouts/th.js]: same extra space here.

File: apps/keyboard/js/layouts/th.js [hunk 7]
```diff
@@ -0,0 +1,134 @@
...
+      { value: 'ใ' }, { value: 'ฝ' },
+      { value: '⌫', keyCode: KeyEvent.DOM_VK_BACK_SPACE }
+    ], [
+      { value: '&nbsp', ratio: 9, keyCode: KeyboardEvent.DOM_VK_SPACE },
+      { value: '↵', ratio: 2, keyCode: KeyEvent.DOM_VK_RETURN }
+    ]
+  ],
+  upperCase: {
+    'ๆ': '฿',
+    'ภ': '1',
```
Comment (RudyLu) [inline apps/keyboard/js/layouts/th.js]: We would have numbers in the symbolLayout (or page[1]), so would not recommend we repeat them here, but I know nothing about Thai, so feel free to keep it if you think this is good enough.

File: apps/keyboard/js/layouts/th.js [hunk 8]
```diff
@@ -0,0 +1,134 @@
...
+      [
+        { value: '!' }, { value: '@' }, { value: '#' }, { value: '$' },
+        { value: '%' }, { value: '^' }, { value: '&' }, { value: '*' },
+        { value: '£' }, { value: '€' }
+      ], [
+        { value: '|' }, { value: '~' }, { value: '\"' }, { value: '`' },
+        { value: '{' }, { value: '}' }, { value: '_' }, { value: '-' },
+        { value: '+' }, { value: '=' }
+      ], [
+        { value: 'กขค'/*กขค*/, ratio: 1.5, keyCode: KeyEvent.DOM_VK_ALT },
```
Comment (RudyLu) [inline apps/keyboard/js/layouts/th.js]: you would need to add `targetPage: 2` to this key, if you want to use this key to switch to a different page.

**Pull request closed without merging**